logicrunner/statestorage.go

```diff
@@ -143,6 +163,45 @@ func (ss *stateStorage) DeleteObjectState(ref insolar.Reference) {
 	delete(ss.state, ref)
 }
 
-func (ss *stateStorage) StateMap() *map[insolar.Reference]*ObjectState {
-	return &ss.state
+func (ss *stateStorage) IsEmpty() bool {
+	return len(ss.state) == 0
+}
+
+func (ss *stateStorage) OnPulse(ctx context.Context, pulse insolar.Pulse) []insolar.Message {
+	ss.Lock()
+	defer ss.Unlock()
+
+	onPulseMessages := make([]insolar.Message, 0)
+
+	oldState := ss.state
+	ss.state = make(map[insolar.Reference]*ObjectState)
+	for objectRef, objectState := range oldState {
+		objectState.Lock()
+
+		meNext, _ := ss.jetCoordinator.IsAuthorized(
+			ctx, insolar.DynamicRoleVirtualExecutor, *objectRef.Record(), pulse.PulseNumber, ss.jetCoordinator.Me(),
+		)
+
+		if broker := objectState.ExecutionBroker; broker != nil {
```

Comment:
maybe more "go way"

would be 
broker := objectState.ExecutionBroker
`
if broker == nil {
    objectState.Unlock()
    continue
}
...
`

so there will be less intendation

Reply:
rewrote